tests/validation/cattlevalidationtest/core/test_machine.py

```diff
@@ -0,0 +1,264 @@
+                                          "region": region,
+                                          "size": size
+                                          }
+                   }
+    # Create a Digital Ocean Machine with invalid access token
+    machine = client.create_machine(**create_args)
+
+    machine = wait_until_expected_state(client, machine, "error")
+    assert machine.state == 'error'
+    assert error_msg_auth_failure in machine.transitioningMessage
+
+    hosts = machine.hosts()
+    assert len(hosts) == 0
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+
+@if_machine_digocean
+def test_digital_ocean_machine_invalid_region(client):
+
+    name = random_str()
+    create_args = {"name": name,
+                   "digitaloceanConfig": {"accessToken": access_key,
+                                          "image": image_name,
+                                          "region": "abc",
+                                          "size": size
+                                          }
+                   }
+    # Create a Digital Ocean Machine with invalid access token
+    machine = client.create_machine(**create_args)
+
+    machine = wait_until_expected_state(client, machine, "error",
+                                        timeout=DEFAULT_TIMEOUT)
+
+    assert machine.state == 'error'
+    assert error_msg_invalid_region in machine.transitioningMessage
+
+    hosts = machine.hosts()
+    assert len(hosts) == 0
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+
+def digital_ocean_machine_life_cycle(client, configs, expected_values):
+
+    # Create a Digital Ocean Machine
+    machine = client.create_machine(**configs)
+
+    machine = client.wait_success(machine, timeout=DEFAULT_TIMEOUT)
+    assert machine.state == 'active'
+
+    # Wait until host shows up with some physicalHostId
+    machine = wait_for_host(client, machine)
+    host = machine.hosts()[0]
+    assert host.state == 'active'
+    assert machine.accountId == host.accountId
+
+    # Check that the droplet that is being created in Digital Ocean has the
+    # correct configurations
+
+    droplet = check_host_in_digital_ocean(host.ipAddresses()[0].address)
+
+    assert droplet is not None
+    assert droplet["name"] == machine.name
+    assert droplet["image"]["slug"] == expected_values["image"]
+    assert droplet["size_slug"] == expected_values["size"]
+    assert droplet["region"]["slug"] == expected_values["region"]
+
+    # Remove the machine and make sure that the host
+    # and the machine get removed
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+    host = client.reload(machine.hosts()[0])
+    assert host.state == 'removed'
+
+    wait_for_host_destroy_in_digital_ocean(host.ipAddresses()[0].address)
+
+
+def wait_for_host(client, machine, timeout=DEFAULT_TIMEOUT):
```

Comment:
Again, can possibly reuse son's function. If it isn't a perfect fit, I'd be fine leaving this as is.